pkg/utils/utils.go

```diff
@@ -184,18 +189,51 @@ func SyncNodeState(newState *sriovnetworkv1.SriovNetworkNodeState) error {
 	return nil
 }
 
-// SkipConfigVf Use systemd service to configure switchdev mode or BF-2 NICs in OpenShift
-func SkipConfigVf(ifSpec sriovnetworkv1.Interface, ifStatus sriovnetworkv1.InterfaceExt) bool {
+// skipConfigVf Use systemd service to configure switchdev mode or BF-2 NICs in OpenShift
+func skipConfigVf(ifSpec sriovnetworkv1.Interface, ifStatus sriovnetworkv1.InterfaceExt) (bool, error) {
 	if ifSpec.EswitchMode == sriovnetworkv1.ESwithModeSwitchDev {
-		glog.V(2).Infof("SkipConfigVf(): skip config VF for switchdev device")
-		return true
+		glog.V(2).Infof("skipConfigVf(): skip config VF for switchdev device")
+		return true, nil
 	}
+
 	// Nvidia_mlx5_MT42822_BlueField-2_integrated_ConnectX-6_Dx in OpenShift
 	if ClusterType == ClusterTypeOpenshift && ifStatus.Vendor == VendorMellanox && ifStatus.DeviceID == DeviceBF2 {
-		glog.V(2).Infof("SkipConfigVf(): skip config VF for BF2 device")
-		return true
+		// TODO: remove this when switch to the systemd configuration support.
+		mode, err := mellanoxBlueFieldMode(ifStatus.PciAddress)
+		if err != nil {
+			return false, fmt.Errorf("failed to read Mellanox Bluefield card mode for %s,%v", ifStatus.PciAddress, err)
+		}
+
+		if mode == bluefieldConnectXMode {
+			return false, nil
+		}
+
+		glog.V(2).Infof("skipConfigVf(): skip config VF for Bluefiled card on DPU mode")
+		return true, nil
 	}
-	return false
+
+	return false, nil
+}
+
+// GetPfsToSkip return a map of devices pci addresses to should be configured via systemd instead if the legacy mode
+// we skip devices in switchdev mode and Bluefield card in ConnectX mode
+func GetPfsToSkip(ns *sriovnetworkv1.SriovNetworkNodeState) (map[string]bool, error) {
+	vfsToSkip := map[string]bool{}
+	for _, ifaceStatus := range ns.Status.Interfaces {
+		for _, iface := range ns.Spec.Interfaces {
+			if iface.PciAddress == ifaceStatus.PciAddress {
+				skip, err := skipConfigVf(iface, ifaceStatus)
+				if err != nil {
+					glog.Errorf("GetPfsToSkip(): fail to check for skip VFs %s: %v.", iface.PciAddress, err)
+					return vfsToSkip, err
+				}
+				vfsToSkip[iface.PciAddress] = skip
+				break
+			}
+		}
+	}
+
+	return vfsToSkip, nil
```

Comment:
Your function is called `GetPfsToSkip` and you're variable is called `vfsToSkip`

Reply:
done thanks!